extensions/cornerstone/src/commandsModule.ts

```diff
@@ -248,18 +249,26 @@ function commandsModule({
     },
     updateStoredPositionPresentation: ({
       viewportId,
-      displaySetInstanceUID,
+      displaySetInstanceUIDs,
       referencedImageId,
       options,
     }) => {
       const presentations = cornerstoneViewportService.getPresentations(viewportId);
       const { positionPresentationStore, setPositionPresentation, getPositionPresentationId } =
         usePositionPresentationStore.getState();
 
-      // Look inside positionPresentationStore and find the key that includes the displaySetInstanceUID
+      // Look inside positionPresentationStore and find the key that includes ALL the displaySetInstanceUIDs
       // and the value has viewportId as activeViewportId.
       const previousReferencedDisplaySetStoreKey = Object.entries(positionPresentationStore).find(
-        ([key, value]) => key.includes(displaySetInstanceUID) && value.viewportId === viewportId
+        ([key, value]) => {
+          if (!displaySetInstanceUIDs || !Array.isArray(displaySetInstanceUIDs)) {
```

Comment:
Can't you put the displaySetInstanceUIDs outside the test, just setting previouseReferencedDSSK to undefined if !displaySetInstanceUIDs?.length >= 0
Otherwise you iterate every time.